bioptim/optimization/optimal_control_program.py

```diff
@@ -380,7 +380,17 @@ def __init__(
         # Prepare the variable mappings
         if variable_mappings is None:
             variable_mappings = BiMappingList()
-        NLP.add(self, "variable_mappings", variable_mappings, True)
+
+        variable_mappings_per_phase = [{} for _ in range(self.n_phases)]
+        for mappings in variable_mappings:
```

Comment:
The test below should be a check method of BiMappingList. This method should be called here though